pkg/services/secrets/kvstore/migrations/to_plugin_mig.go

```diff
@@ -58,22 +55,32 @@ func (s *MigrateToPluginService) Migrate(ctx context.Context) error {
 			logger.Warn("unable to determine whether plugin startup failures are fatal - continuing migration anyway.")
 		}
 
+		// get all secrets in the fallback store
 		allSec, err := fallbackStore.GetAll(ctx)
 		if err != nil {
 			return nil
 		}
 		totalSec := len(allSec)
-		// We just set it again as the current secret store should be the plugin secret
 		logger.Debug(fmt.Sprintf("Total amount of secrets to migrate: %d", totalSec))
-		for i, sec := range allSec {
-			logger.Debug(fmt.Sprintf("Migrating secret %d of %d", i+1, totalSec), "current", i+1, "secretCount", totalSec)
-			err = s.secretsStore.Set(ctx, *sec.OrgId, *sec.Namespace, *sec.Type, sec.Value)
-			if err != nil {
-				return err
+
+		// during migration we need to have fallback enabled while we move secrets to plugin
+		err = pluginStore.WithFallbackEnabled(func() error {
```

Comment:
Do you think this should also encompass the part where it queries the fallback store? Probably not much difference but I don't think it can hurt

Reply:
Done on https://github.com/grafana/grafana/pull/54496/commits/92d399be2f4589b99ced66ccba1f0e5b2548cb74